src/plugins/azgenerator/TemplateAzureCliCustom.ts

```diff
@@ -128,72 +133,68 @@ function GetCommandBody(model: CodeModelAz, required: boolean, needUpdate: boole
-            let methodName: string = model.Command_MethodName;
-
-            if (methodName != "show" && methodName != "list" && methodName != "delete") {
-                // body transformation
-                if (!isUpdate) {
-                    output_body.push("    body = {}");
-                }
-                else {
-                    // XXX-FIX-GET
-                    //if (methods.indexOf("show") >= 0)
-                    //{
-                    //    model.SelectCommand("show");
-                    //    output_body.push("    body = " + GetMethodCall(model) + ".as_dict()");
-                    //}
-                    //else
-                    //{
-                    output_body.push("    body = {}");
-                    //}
-                }
+        // create body transformation for methods that support it
+        let methodName: string = model.Command_MethodName;
 
-                if (model.SelectFirstMethodParameter()) {
-                    do {
-                        let access = "    body"
-                        if (model.MethodParameter_In == ParameterLocation.Body) {
-                            let parts = model.Option_PathSdk.split("/");
-                            let last: string = parts.pop();
-                            parts.forEach(part => {
-                                if (part != "" && part != "*") {
-                                    access += ".setdefault('" + part + "', {})";
-                                }
-                            });
-
-                            access += "['" + last + "'] = ";
+        if (methodName != "show" && methodName != "list" && methodName != "delete") {
+            // body transformation
 
-                            if (model.MethodParameter_IsList) {
-                                if (model.MethodParameter_Type != SchemaType.Dictionary) {
-                                    // a comma separated list
-                                    access += "None if " + model.MethodParameter_Name + " is None else " + model.MethodParameter_Name;
+            if (model.SelectFirstMethodParameter()) {
+                do {
+                    if(model.MethodParameter_IsFlattened) {
+                        let bodyName = model.MethodParameter_Name;
+                        output_body.push("    " + bodyName + " = {}");
+                        let body = model.MethodParameter;
+
+                        while(model.SelectNextMethodParameter()) {
+                            let access = "    " + bodyName;
+                            let param = model.MethodParameter;
+                            let oriParam = (param['originalParameter']);
+                            if(oriParam == body) {
+                                if(param['pathToProperty']?.length == 1) {
+                                    let pathParam = param['pathToProperty'][0];
+                                    access += ".setdefault('" + pathParam.language['python'].name + "', {})";
```

Comment:
>".setdefault('" + pathParam.language['python'].name + "', {})" [](start = 46, length = 62)

from curious, any reason not to use `` ?

Reply:
This is copied from the original custom template. I changed it now